tmt/base.py

```diff
@@ -448,37 +455,39 @@ def from_spec(cls, raw: _RawRequireFile) -> 'RequireFile':
     @staticmethod
     def validate() -> Tuple[bool, str]:
         """
-        Validate file requirement and return a human readable error
+        Validate file dependency and return a human readable error
 
         There is no way to check validity of type or pattern string at
         this time. Return a tuple (boolean, message) as the result of
         validation. The boolean specifies the validation result and the
-        message the validation error. In case the file requirement is
+        message the validation error. In case the file dependency is
         valid, return an empty string as the message.
         """
         return True, ''
 
 
-_RawRequireItem = Union[str, _RawRequireFmfId, _RawRequireFile]
+_RawRequireItem = Union[str, _RawDependencyFmfId, _RawDependencyFile]
 _RawRequire = Union[_RawRequireItem, List[_RawRequireItem]]
```

Comment:
Shouldn't these two be `_RawDependencyItem` and `_RawDependency`? Or are we keeping those intentionally with the `Require` wording?

Reply:
Yes, these were left with `Require` on purpose, to follow the chain of `require` => `_RawRequire` => `_RawRequireItem`, with "raw" relating to "raw" input from fmf data/command line used for `_RawAdjust`, `_RawAdjustItem`, etc.

Reply:
I completely get the "raw" part and it makes sense. However, the chain explanation did not help much as `_RawRequire` is used for both `require` and `recommend`:

```python
    def _normalize_require(
            self,
            value: Optional[_RawRequire],
            logger: tmt.log.Logger) -> List[Dependency]:
        return normalize_require(value, logger)

    def _normalize_recommend(
            self,
            value: Optional[_RawRequire],
            logger: tmt.log.Logger) -> List[Dependency]:
        return normalize_require(value, logger)
```

Reply:
Hm, that's true... Ok, let me turn `_RawRequire` to `_RawDependency` then.

Reply:
Needs a rebase anyway.

Reply:
Good, thanks!